pkg/openshift-routes/main.go

```diff
@@ -0,0 +1,30 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package platform contains the availability of the OpenShift Routes API.
+package openshift_routes
+
+// Platform holds the auto-detected platform type.
+type OpenShiftRoutesAvailability bool
```

Comment:
Could we use an `int` instead of `bool` to be prepared in case of any future extensions?

Reply:
I don't see the value of the change... something is available, not available or its availability is not known. But I'll change it if you think is important